src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -212,6 +238,77 @@ public static void Unregister()
+        ///     Path to the directory containing the DotNet SDK.
+        /// </param>
+        private static void ApplyDotNetSdkEnvironmentVariables(string dotNetSdkPath)
+        {
+            const string MSBUILD_EXE_PATH = nameof(MSBUILD_EXE_PATH);
+            const string MSBuildExtensionsPath = nameof(MSBuildExtensionsPath);
+            const string MSBuildSDKsPath = nameof(MSBuildSDKsPath);
+
+            var variables = new Dictionary<string, string>
+            {
+                [MSBUILD_EXE_PATH] = dotNetSdkPath + "MSBuild.dll",
+                [MSBuildExtensionsPath] = dotNetSdkPath,
+                [MSBuildSDKsPath] = dotNetSdkPath + "Sdks"
+            };
+
+            foreach (var kvp in variables)
+            {
+                Environment.SetEnvironmentVariable(kvp.Key, kvp.Value);
+            }
+        }
+
+        /// <summary>
+        ///     Load Microsoft.Build core dlls into the current AppDomain from the specified path.
+        /// </summary>
+        /// <param name="msbuildPath">
+        ///     Path to the directory containing a deployment of MSBuild binaries.
+        ///     A minimal MSBuild deployment would be the publish result of the Microsoft.Build.Runtime package.
+        ///
+        ///     In order to restore and build real projects, one needs a deployment that contains the rest of the toolchain (nuget, compilers, etc.).
+        ///     Such deployments can be found in installations such as Visual Studio or dotnet CLI.
+        /// </param>
+        public static void LoadFromMSBuildPath(string msbuildPath)
+        {
+            if (string.IsNullOrWhiteSpace(msbuildPath))
+            {
+                throw new ArgumentException("Value may not be null or whitespace", nameof(msbuildPath));
+            }
+
+            if (!Directory.Exists(msbuildPath))
+            {
+                throw new ArgumentException($"Directory \"{msbuildPath}\" does not exist", nameof(msbuildPath));
+            }
+
+            if (!CanRegister)
+            {
+                var loadedAssemblyList = string.Join(Environment.NewLine, LoadedMsBuildAssemblies.Select(a => a.GetName()));
+
+                var error = $"{typeof(MSBuildLocator)}.{nameof(RegisterInstance)} was called, but MSBuild assemblies were already loaded." +
+                    Environment.NewLine +
+                    $"Ensure that {nameof(RegisterInstance)} is called before any method that directly references types in the Microsoft.Build namespace has been called." +
+                    Environment.NewLine +
+                    "Loaded MSBuild assemblies: " +
+                    loadedAssemblyList;
+
+                throw new InvalidOperationException(error);
+            }
+
+            foreach (var msBuildAssembly in s_msBuildAssemblies)
+            {
+                var targetAssembly = Path.Combine(msbuildPath, msBuildAssembly + ".dll");
+                Assembly.LoadFrom(targetAssembly);
```

Comment:
I'm not sure about how LoadFrom works on Core, but I vaguely recall that it might require the assembly to be present in deps.json. If so, that might still be mostly ok because the use case is to bind static compilation against msbuild to a dynamic location. 

But I'm unsure how this would behave if you had an indirection from a main app that knew nothing of msbuild to a plug-in that did. It seems you would want to resolve msbuild into the caller's assemblyloadcontext. I don't know if this accomplishes that or not.

@jeffschwMSFT and/or team should review this.